src/webgpu/web_platform/canvas/getCurrentTexture.spec.ts

```diff
@@ -2,11 +2,243 @@ export const description = `
+      colorAttachments: [
+        {
+          view: frameTexture.createView(),
+          clearValue: [1.0, 0.0, 0.0, 1.0],
+          loadOp: 'clear',
+          storeOp: 'store',
+        },
+      ],
+    });
+    pass.end();
+    t.device.queue.submit([encoder.finish()]);
+
+    // Calling getCurrentTexture after performing some work on the texture returns the same texture.
+    t.expect(frameTexture === ctx.getCurrentTexture());
+
+    // Ensure that getCurrentTexture does not clear the texture.
+    t.expectSingleColor(frameTexture, frameTexture.format, {
+      size: [frameTexture.width, frameTexture.height, 1],
+      exp: { R: 1, G: 0, B: 0, A: 1 },
+    });
+
+    frameTexture.destroy();
+
+    // Calling getCurrentTexture after destroying the texture still returns the same texture.
+    t.expect(frameTexture === ctx.getCurrentTexture());
+  });
 
 g.test('multiple_frames')
-  .desc(`Checks the value of getCurrentTexture within one frame and across multiple frames.`)
-  .unimplemented();
+  .desc(`Checks that the value of getCurrentTexture differs across multiple frames.`)
+  .params(u =>
+    u //
+      .combine('canvasType', kAllCanvasTypes)
+      .beginSubcases()
+      .combine('clearTexture', [true, false])
+  )
+  .fn(async t => {
+    const { canvasType, clearTexture } = t.params;
+
+    return new Promise(resolve => {
+      const ctx = t.initCanvasContext(canvasType);
+      let prevTexture: GPUTexture;
+      let frameCount = 0;
+
+      async function frameCheck() {
+        const currentTexture = ctx.getCurrentTexture();
+
+        if (prevTexture) {
+          // Ensure that each frame a new texture object is returned.
+          t.expect(currentTexture !== prevTexture);
+
+          // Ensure that texture contents are transparent black.
+          t.expectSingleColor(currentTexture, currentTexture.format, {
+            size: [currentTexture.width, currentTexture.height, 1],
+            exp: { R: 0, G: 0, B: 0, A: 0 },
+          });
+        }
+
+        if (clearTexture) {
+          // Clear the texture to test that texture contents don't carry over from frame to frame.
+          const encoder = t.device.createCommandEncoder();
+          const pass = encoder.beginRenderPass({
+            colorAttachments: [
+              {
+                view: currentTexture.createView(),
+                clearValue: [1.0, 0.0, 0.0, 1.0],
+                loadOp: 'clear',
+                storeOp: 'store',
+              },
+            ],
+          });
+          pass.end();
+          t.device.queue.submit([encoder.finish()]);
+        }
+
+        prevTexture = currentTexture;
+
+        if (frameCount++ < 5) {
+          // Which method will be used to begin a new "frame"?
+          switch (canvasType) {
+            case 'onscreen':
+              requestAnimationFrame(frameCheck);
+              break;
+            case 'offscreen': {
+              (ctx.canvas as OffscreenCanvas).transferToImageBitmap();
```

Comment:
If transferToImageBitmap isn't available we should skip the offscreen case (unless you can think of another way to test it)